README.md
# serie
soccer tournament simulator
follow the instructions in the game manu


history.txt
0
team      round score goal  fan
firenze   14    32    26    108
marsiglia 14    22    26    98
milano    14    20    24    94
turino    14    20    22    92
genoa     14    17    19    86
tunisia   14    17    18    85
venezia   14    16    19    85
bologna   14    10    15    75
name           goal   team
keegan         6      milano
bergkemp       5      firenze
ronaldo        4      milano
rivera         4      turino
careca         4      turino
donadoni       4      turino
zico           4      venezia
socrates       4      venezia
zidane         4      marsiglia
matthaeus      4      marsiglia
name           assist team
desaily        7      marsiglia
p.rossi        6      firenze
keegan         5      milano
hurst          5      milano
daglish        4      genoa
beckenbauer    4      tunisia
garrincha      4      venezia
vieri          4      bologna
r.baggio       4      venezia
cabrini        4      firenze
name           score  team
kopa           65      bologna
g.muller       64      turino
keegan         64      milano
rivera         63      turino
raul           62      turino
careca         62      turino
donadoni       61      turino
ronaldo        61      milano
garrincha      61      venezia
b.chalton      61      bologna
player       goal assist score
kahn           0     0     5.800000000000001

cantona        3     1     5.800000000000001

ronaldo        4     4     6.1000000000000005

rijkaard       1     1     6.0

facchetti      2     1     5.9

papin          4     2     6.0

keegan         6     5     6.4

bierhoff       0     2     5.9

hurst          1     5     5.800000000000001

kaka           2     1     5.5

blanc          1     2     5.7

score 20 goal 24 number of fans 94
buffon         0     0     6.0

eusebio        1     4     6.0

rivera         4     3     6.300000000000001

raul           0     2     6.2

g.muller       2     4     6.4

henry          1     0     5.6000000000000005

careca         4     3     6.2

conti          2     1     5.6000000000000005

gascoigne      2     2     5.800000000000001

donadoni       4     2     6.1000000000000005

cassano        2     1     5.800000000000001

score 20 goal 22 number of fans 92
zoff           0     0     5.7

gullit         2     2     5.800000000000001

puskas         0     3     5.7

b.moore        1     1     6.0

baresi         2     4     5.7

owen           2     1     5.9

p.falcao       3     0     5.800000000000001

daglish        2     4     5.9

shevchenko     3     2     6.0

deisler        1     1     6.0

stam           3     1     5.9

score 17 goal 19 number of fans 86
chilavert      0     0     5.7

cruyff         1     1     5.7

v.basten       1     2     5.7

garrincha      2     4     6.1000000000000005

zico           4     3     5.800000000000001

stoichkov      1     0     5.800000000000001

kempes         3     1     5.7

socrates       4     1     5.9

r.baggio       2     4     6.0

totti          1     1     5.7

g.weah         0     2     5.800000000000001

score 16 goal 19 number of fans 85
zenga          0     0     5.6000000000000005

platini        1     3     5.7

nedved         3     2     6.0

zidane         4     3     5.9

m.laudrup      3     1     5.9

matthaeus      4     1     5.800000000000001

hagi           1     3     5.800000000000001

desaily        4     7     6.0

h.shanchez     0     2     5.5

maldini        4     2     6.0

r.keane        2     2     5.800000000000001

score 22 goal 26 number of fans 98
yashin         0     0     5.5

beckenbauer    2     4     5.5

ronaldinho     2     2     5.300000000000001

r.koeman       1     1     5.4

tostao         0     1     5.4

rensenbrink    2     2     5.6000000000000005

ballack        3     0     5.4

crespo         0     1     5.300000000000001

v.nistelrooy   4     3     5.7

lampard        2     2     5.6000000000000005

pele           2     2     5.6000000000000005

score 17 goal 18 number of fans 85
banks          0     0     5.7

d.stifano      1     1     5.9

b.chalton      0     0     6.1000000000000005

g.best         2     1     5.7

kopa           1     1     6.5

vialli         2     2     6.0

neeskens       2     3     6.0

mancini        0     2     5.5

trezeguet      3     0     5.7

b.laudrup      2     1     5.800000000000001

vieri          2     4     5.9

score 10 goal 15 number of fans 75
schmeichel     0     0     5.5

maradona       2     3     5.7

beckham        3     1     5.6000000000000005

klinsman       0     0     5.2

didi           3     3     5.800000000000001

brehme         4     1     5.6000000000000005

batistuta      2     3     5.6000000000000005

cabrini        2     4     5.5

p.rossi        2     6     5.9

canavaro       3     3     5.6000000000000005

bergkemp       5     2     5.800000000000001

score 32 goal 26 number of fans 108
papin from milano to turino
rivera from turino to milano
daglish from genoa to venezia
stoichkov from venezia to genoa
m.laudrup from marsiglia to tunisia
ballack from tunisia to marsiglia
neeskens from bologna to firenze
canavaro from firenze to bologna
#########################################################################
########################################################################
new season
#####################################################################
###################################################################
1
team      round score goal  fan
tunisia   14    27    23    92
marsiglia 14    26    24    99
turino    14    22    25    93
bologna   14    21    22    80
milano    14    17    18    82
venezia   14    16    14    72
firenze   14    15    14    83
genoa     14    10    15    68
name           goal   team
g.muller       7      turino
blanc          6      milano
canavaro       5      bologna
mancini        4      bologna
pele           4      tunisia
papin          4      turino
ballack        4      marsiglia
careca         3      turino
m.laudrup      3      tunisia

src/tournament/team.java
package tournament;
class team
{
  String name;
  player p=new player();
  player[] member={p,p,p,p,p,p,p,p,p,p,p};
  int score;
  int round;
  int goal;
  int fan;
  int igoal;
    int cupgoal;
    int cupgamegoal;
    int cupid;
    int icontrol;
    int ishoot;
    int isshoot;
  public team()
  {
  }
  public void add(player p,int i)
  {
    member[i]=p;
  }
  public String getinf()
  {
    String s="";
    for(int j=0;j<11;j++)
    {
      s=s+member[j].getinf();
      //s=s+"\n";
    }
    s=s+"score "+score+" goal "+goal+" number of fans "+fan;
    return s;
  }
    public void arrange(){
        for(int i=1;i<4;i++){
            for(int j=i;j<11;j++){
                if(member[i].defence/(member[i].pass+member[i].shoot+0.1)<member[j].defence/(member[j].pass+member[j].shoot+0.1)){
                    player tmp=member[i];
                    member[i]=member[j];
                    member[j]=tmp;
                }
            }
        }
        for(int i=4;i<7;i++){
            for(int j=i;j<11;j++){
                if(member[i].pass/(member[i].shoot+0.1)<member[j].pass/(member[j].shoot+0.1)){
                    player tmp=member[i];
                    member[i]=member[j];
                    member[j]=tmp;
                }
            }
        }
        rearrange();
    }
    public void rearrange(){
        int i=(int)(Math.random()*3)+1;
        int j=(int)(Math.random()*10)+1;
        if(member[i].defence/(member[i].shoot+member[i].pass+0.1)<member[j].defence/(member[j].shoot+member[j].pass+0.1)){
            player tmp=member[i];
            member[i]=member[j];
            member[j]=tmp;
        }

        i=(int)(Math.random()*3)+4;
        j=(int)(Math.random()*10)+1;
        if(member[i].pass/(member[i].shoot+member[i].defence+0.1)<member[j].pass/(member[j].shoot+member[j].defence+0.1)){
            player tmp=member[i];
            member[i]=member[j];
            member[j]=tmp;
        }

        i=(int)(Math.random()*4)+7;
        j=(int)(Math.random()*10)+1;
        if(member[i].shoot/(member[i].pass+member[i].defence+0.1)<member[j].shoot/(member[j].pass+member[j].defence+0.1)){
            player tmp=member[i];
            member[i]=member[j];
            member[j]=tmp;
        }
    }
    public void printFormation(){
        System.out.println(name);
        System.out.println(member[0].name);
        System.out.println(member[1].name+"/"+member[2].name+"/"+member[3].name);
        System.out.println(member[4].name+"/"+member[5].name+"/"+member[6].name);
        System.out.println(member[7].name+"/"+member[8].name+"/"+member[9].name+"/"+member[10].name);
    }
    public player randomp(int pos){//0:b;1:m;2:f
        int i;
        switch (pos){
            case 0:
                i=(int)(Math.random()*13)+1;
                if(i>10){
                    i-=10;
                }
                return member[i];
            case 1:
                i=(int)(Math.random()*13)+1;
                if(i>10){
                    i-=7;
                }
                return member[i];
            case 2:
                i=(int)(Math.random()*14)+1;
                if(i>10){
                    i-=4;
                }
                return member[i];
            default:
                return null;
        }
    }
    public void setZeroBM(){
        icontrol=0;
        ishoot=0;
        isshoot=0;
        for(player p:member){
            p.ipass=0;
            p.ispass=0;
            p.itackle=0;
            p.istackle=0;
            p.ishoot=0;
            p.isshoot=0;
            p.iassist=0;
            p.igoal=0;
        }
    }
}

src/tournament/Jieshuo.java
package tournament;

import java.util.Random;
class Jieshuo
{
  String[] cbd;
  String[] czj;
  String[] czjj;
  String[] czjc;
  String[] cbr;
  String[] bd;
  String[] dmj;
  String[] bp;
  String[] j;
  public Jieshuo()
  {
    cbd=new String[]{" get the ball back"};//"英勇地抢断","将球断下","阻止了对方的进攻","一个飞铲将球断下","夺回球权"};
    czj=new String[]{" win a penalty"};//"在禁区内被放倒","赢得了点球","倒在了禁区内"
      //,"点球!点球!点球!他立功了，他立功了!不要给对方任何的机会。 伟大夏越队的左后卫!他继承了夏越的光荣的传统。法切蒂、卡布里尼、马尔蒂尼在这一刻灵魂附体!他一个人他代表了夏越足球悠久的历史和传统，在这一刻他不是一个人在战斗，他不是一个人! "
      //,"这是个假摔啊，裁判竟然给了点球"};
    czjj=new String[]{" shoot in"};//"把球罚进死角","踢出勺子点球","骗过了门将","射门，球打在门将身上弹进球门","球进啦!比赛结束了!伟大的夏越!伟大的夏越的左后卫!马尔蒂尼今天生日快乐!夏越万岁! "};;
    czjc=new String[]{" save"};//"门将跳起了面条舞，对方果然打飞了","一个鱼跃把球扑出","用意念让球飞出","轻松把球得到"};
    cbr=new String[]{" pass"};//"过顶长传","送出直塞","挑传给队友","底线传中","将球停给队友","将球球交给队友","开出任意球","开出角球","精彩的二过一"};
    bd=new String[]{" defend"};//"舍身堵抢眼","干净地将球断下","大脚解围","头球解围","将球破坏出底线","成功的抢断","一个飞铲把球断下"};
    dmj=new String[]{" shoot"};//"晃过了所有人","杀入禁区，射了","长途奔袭，直接面对门将，打门","飞身垫射","临空抽射","倒钩射门","拉出空当，射门","在禁区外起脚远射","圆月弯刀","踢出电梯球","回头望月","没踢中足球，禁区内一片混乱","传球，结果变成了一次射门","将球吊向球门",
      //"超远距离射门","门前抢点","补射","没人防守，射门","和队友打出精妙配合，射门","生吃对方后卫，射门"};
    bp=new String[]{" save"};//"把球吓飞了","飞身将球扑出","单掌将球托出","双拳将球击出","轻松将球得到","将射门的球员吓傻了","奋不顾身的将球得到"};
    j=new String[]{" goal"};//"将球打进","穿裆将球打进","把球打进死角","进了，这球其实是个乌龙","进球了"};
  }
  public String cbd(String n)
  {
    Random r = new Random();
   return n+cbd[(int)(cbd.length* r.nextDouble())];
  }
  public String czj(String n)
  {
    Random r = new Random();
   return n+czj[(int)(czj.length* r.nextDouble())];
  }
  public String czjj(String n)
  {
    Random r = new Random();
   return n+czjj[(int)(czjj.length* r.nextDouble())];
  }
  public String czjc(String n)
  {
    Random r = new Random();
   return n+czjc[(int)(czjc.length* r.nextDouble())];
  }
  public String cbr(String n)
  {
    Random r = new Random();
   return n+cbr[(int)(cbr.length* r.nextDouble())];
  }
  public String bd(String n)
  {
    Random r = new Random();
   return n+bd[(int)(bd.length* r.nextDouble())];
  }
  public String dmj(String n)
  {
    Random r = new Random();
   return n+dmj[(int)(dmj.length* r.nextDouble())];
  }
  public String bp(String n)
  {
    Random r = new Random();
   return n+bp[(int)(bp.length* r.nextDouble())];
  }
  public String j(String n)
  {
    Random r = new Random();
   return n+j[(int)(j.length* r.nextDouble())];
  }
}

src/tournament/Playerlist.java
package tournament;
class playerlist
{
  player[] p;
  public playerlist()
  {
    p=new player[132];
  }
  public void generatelist(team[] t)
  {
    for(int i=0;i<12;i++)
    {
      for(int j=0;j<11;j++)
      {
        p[11*i+j]=t[i].member[j];
      }
    }
  }
  public void sort(boolean a,boolean b)
//tt:goal   ft:saaist    ff:score
  {
    if(a)//goal
    {
      int min=0; player temp=new player();
      for(int i=0;i<88;i++)
      {
        min=i;
        for (int j=i+1;j<88;j++){
          if(p[j].goal>p[i].goal)
          {
            temp=p[i];
            p[i]=p[j];
            p[j]=temp;
          }
        }
      }
    }
    else if(b)//assist
    {
      int min=0; player temp=new player();
      for(int i=0;i<88;i++)
      {
        min=i;
        for (int j=i+1;j<88;j++){
          if(p[j].assist>p[i].assist)
          {
            temp=p[i];
            p[i]=p[j];
            p[j]=temp;
          }
        }
      }
    }
    else//score
    {
      int min=0; player temp=new player();
      for(int i=0;i<88;i++)
      {
        min=i;
        for (int j=i+1;j<88;j++){
          if(p[j].score>p[i].score)
          {
            temp=p[i];
            p[i]=p[j];
            p[j]=temp;
          }
        }
      }
    }
  }
  //tt:goal   ft:saaist    ff:score
  public String getlist(boolean a,boolean b)
  {
    String s="";
    if(a)//goal
    {
      s=s+"name           goal   team\n";
      for(int i=0;i<10;i++)
      {
        s=s+p[i].name;
        for(int j=0;j<15-p[i].name.length();j++)
        {
          s=s+" ";
        }
        s=s+p[i].goal;
        s=s+"      "+p[i].team+"\n";
      }
      return s;
    }
    else if(b)//assist
    {
      s=s+"name           assist team\n";
      for(int i=0;i<10;i++)
      {
        s=s+p[i].name;
        for(int j=0;j<15-p[i].name.length();j++)
        {
          s=s+" ";
        }
        s=s+p[i].assist;
        s=s+"      "+p[i].team+"\n";
      }
      return s;
    }
    else//score
    {
      s=s+"name           score  team\n";
      for(int i=0;i<10;i++)
      {
        s=s+p[i].name;
        for(int j=0;j<15-p[i].name.length();j++)
        {
          s=s+" ";
        }
        s=s+p[i].score/100.0;
        s=s+"      "+p[i].team+"\n";
      }
      return s;
    }
  }
  public player findplayer(String s)
 {
   for(int i=131;i>-1;i--)
   {
     if(p[i].name.equals(s))
     {
       return p[i];
     }
     else
     {
       continue;
     }
   }
   System.out.println("can't find");
   return null;
 }
  public void set0()
 {
   for(int i=0;i<132;i++)
   {
     p[i].score=0;
     p[i].assist=0;
     p[i].goal=0;
   }
 }
}

src/tournament/tournament.java
package tournament;

import java.io.*;
class tournament
{
  public static teamlist tl=new teamlist(1,1);
  public static playerlist pl=new playerlist();
  public static void main(String[] args)
  { 
    readteam();
    readplayer();
    tl.generatera();
    pl.generatelist(tl.ca);

      //tl.changefan();
      /*tl.round=1;
      tl.game=1;
      tl.regroup();
      pl.set0();
      tl.set0();

      writeteam();*/

      /*tl.round=14;
      tl.game=5;*/

    run();
    //if (round==14)
    //{
      //write file to perm(true)
    //}
    //else
    //{
      //write file to temp and inseason
    //rearrange
    writeplayer();
    writeteam();
    //}
  }
  /////////////////////////////////////////////////////////////////////////////////////operator///////////////////////////////////////////////////////
  public static void run()
  {
    System.out.println("1.play next game");
    System.out.println("2.view tournament standing");
    System.out.println("3.view personal standing");
      System.out.println("4.view cup standing");
      System.out.println("5.view second division standing");
    System.out.println("6.view team information");
    System.out.println("7.view personal information");
    System.out.println("8.exit");
    System.out.println("9.next season");
    java.util.Scanner ff=new java.util.Scanner(System.in);
    try{int reply=ff.nextInt();
    String reply2;
      switch(reply)
      {
        case 1: 
        {
          if(tl.round==0)
          {
            System.out.println("season ends");
          }
          else
          {
            team a=tl.chooseteam(tl.round,tl.game,true);
            team b=tl.chooseteam(tl.round,tl.game,false);
              a.arrange();
              b.arrange();
              a.printFormation();
              b.printFormation();
              a.setZeroBM();
              b.setZeroBM();
            tl.playnextgame();
            writegamerecord(a,b);
            writeplayer();
            writeteam();
            run();
            break;
          }
        }
        case 2: 
        {
          //print team rank
          tl.generatera();
          System.out.println(tl.returnrank());
          run();
            break;
        }
        case 3: 
        {
          //print score assist mark
          pl.sort(true,true);
          System.out.println(pl.getlist(true,true));
          pl.sort(false,true);
          System.out.println(pl.getlist(false,true));
          pl.sort(false,false);
          System.out.println(pl.getlist(false,false));
          run();
          break;          
        }
        case 4:{
            for(team t:tl.cu){
                System.out.println(t.name+"    total: "+t.cupgoal+" first round: "+(t.cupgamegoal-1));
            }
            run();
            break;
        }
          case 5:{
              tl.generatera();
              System.out.println(tl.returnSecondRank());
              run();
              break;
          }
        case 6:
        {
          //print team inf
          System.out.println("enter team name");
          java.util.Scanner f=new java.util.Scanner(System.in);
          reply2=f.nextLine();
          System.out.println("player       goal assist score");
          System.out.println(tl.findteam(reply2).getinf());
          run();
          break;
        }
        case 7:
        {
          //print player inf
          System.out.println("enter player name");
          java.util.Scanner f=new java.util.Scanner(System.in);
          reply2=f.nextLine();
          System.out.println("player       goal assist score");
          System.out.print(pl.findplayer(reply2).getinf());
          System.out.println("now in "+pl.findplayer(reply2).team);
          run();
          break;
        }
        case 8:
        {
          //exit
          
          break;
        }
        case 9:
        {
          //new season set 0 write history
          System.out.println("are you sure to end this season");
          System.out.println("0.yes");
          System.out.println("1.no");
          reply=ff.nextInt();
          if(reply==0)
          {
          tl.changefan();
          writehistory();
          tl.round=1;
          tl.game=1;
          tl.year++;
          tl.regroup();
          pl.set0();
          tl.set0();
              writeteam();
              writeplayer();
          }
          run();
          break;
        }
        default:break;
      }
      }catch(Exception ee){ System.out.println(ee);run();}
    //}
  }
  //////////////////////////////////////////////////////////////////////////////////////////read/////////////////////////////////////////////////////////////////////////////
  public static void readteam(){
        
        try{
            BufferedReader br = new BufferedReader(new FileReader("team.txt"));
            String s;
            team t;
            for(int i=0;i<12;i++)
            {
              t=new team();
              s = br.readLine();
              t.name=s;
              s = br.readLine();
              t.score=toint(s);
              s = br.readLine();
              t.round=toint(s);
              s = br.readLine();
              t.goal=toint(s);
              s = br.readLine();
              t.fan=toint(s);
                s = br.readLine();
                t.cupgoal=toint(s);
                s = br.readLine();
                t.cupgamegoal=toint(s);
                s = br.readLine();
                t.cupid=toint(s);
              tl.ca[i]=t;
                //System.out.println(t.name);
              }
            s = br.readLine();
            tl.round=toint(s);
            s = br.readLine();
            tl.game=toint(s);
            s = br.readLine();
            tl.year=toint(s);
            br.close();            
        }
        catch(Exception e){
            System.out.println(e);
        }
        
    }
  public static void readplayer(){
        
        try{
            BufferedReader br = new BufferedReader(new FileReader("player.txt"));
            String s;
            player p;
            for(int i=0;i<12;i++)
            {
              for(int j=0;j<11;j++)
              {
                p=new player();
                s = br.readLine();
                p.name=s;
                s = br.readLine();
                p.shoot=toint(s);
                s = br.readLine();
                p.pass=toint(s);
                s = br.readLine();
                p.defence=toint(s);
                s = br.readLine();
                p.goal=toint(s);
                s = br.readLine();
                p.assist=toint(s);
                s = br.readLine();
                p.save=toint(s);
                s = br.readLine();
                p.score=toint(s);
                s = br.readLine();
                p.team=s;
                tl.ca[i].add(p,j);
                p.team=tl.ca[i].name;
              }
            }

            br.close();            
        }
        catch(Exception e){
            System.out.println(e);
        }
        
    }
  /////////////////////////////////////////////////////////////////////////////////////////write////////////////////////////////////////////////////////////////
  public static void writeplayer()
  {
     try
        {
          BufferedWriter bw = new BufferedWriter(new FileWriter("player.txt"));
          for(int i=0;i<12;i++)
          {
            for(int j=0;j<11;j++)
            {
              bw.write(tl.ca[i].member[j].name);
              bw.newLine();
              bw.write(tostring(tl.ca[i].member[j].shoot));
              bw.newLine();
              bw.write(tostring(tl.ca[i].member[j].pass));
              bw.newLine();
              bw.write(tostring(tl.ca[i].member[j].defence));
              bw.newLine();
              bw.write(tostring(tl.ca[i].member[j].goal));
              bw.newLine();
              bw.write(tostring(tl.ca[i].member[j].assist));
              bw.newLine();
              bw.write(tostring(tl.ca[i].member[j].save));
              bw.newLine();
              bw.write(tostring(tl.ca[i].member[j].score));
              bw.newLine();
              bw.write(tl.ca[i].member[j].team);
              bw.newLine();
            }
          }
          bw.close();
        }
        
        catch(Exception ee){
            System.out.println(ee);
        }
  }
  public static void writeteam()
  {
    try
        {
          BufferedWriter bw = new BufferedWriter(new FileWriter("team.txt"));
           for(int i=0;i<12;i++)
            {
              bw.write(tl.ca[i].name);
              bw.newLine();
              bw.write(tostring(tl.ca[i].score));
              bw.newLine();
              bw.write(tostring(tl.ca[i].round));
              bw.newLine();
              bw.write(tostring(tl.ca[i].goal));
              bw.newLine();
              bw.write(tostring(tl.ca[i].fan));
              bw.newLine();
                bw.write(tostring(tl.ca[i].cupgoal));
                bw.newLine();
                bw.write(tostring(tl.ca[i].cupgamegoal));
                bw.newLine();
                bw.write(tostring(tl.ca[i].cupid));
                bw.newLine();
            }
           bw.write(tostring(tl.round));
           bw.newLine();
           bw.write(tostring(tl.game));
           bw.newLine();
           bw.write(tostring(tl.year));
           bw.newLine();
           bw.close();
        }
        
        catch(Exception ee){
            System.out.println(ee);
        }
  }
  public static void writehistory()
  {
    try
        {
          BufferedWriter bw = new BufferedWriter(new FileWriter("history.txt",true));
          bw.write (tostring(tl.year));
          bw.newLine();
          bw.write ("team      round score goal  fan");
          bw.newLine();
          for(int i=0;i<12;i++)
          {
            String s="";
            s=s+tl.ra[i].name;
            for(int j=0;j<10-tl.ra[i].name.length();j++)
            {
              s=s+" ";
            }
            s=s+tl.ra[i].round;
            for(int j=0;j<6-(""+tl.ra[i].round).length();j++)
            {
              s=s+" ";
            }
            s=s+tl.ra[i].score;
            for(int j=0;j<6-(""+tl.ra[i].score).length();j++)
            {
              s=s+" ";
            }
            s=s+tl.ra[i].goal;
            for(int j=0;j<6-(""+tl.ra[i].goal).length();j++)
            {
              s=s+" ";
            }
            s=s+tl.ra[i].fan;
            bw.write (s);
            bw.newLine();
          }
          pl.sort(true,true);
          bw.write ("name           goal   team");
          bw.newLine();
          for(int i=0;i<10;i++)
          {
            String s="";
            s=s+pl.p[i].name;
            for(int j=0;j<15-pl.p[i].name.length();j++)
            {
              s=s+" ";
            }
            s=s+pl.p[i].goal;
            s=s+"      "+pl.p[i].team;
            bw.write (s);
            bw.newLine();
          }
          pl.sort(false,true);
          bw.write ("name           assist team");
          bw.newLine();
          for(int i=0;i<10;i++)
          {
            String s="";
            s=s+pl.p[i].name;
            for(int j=0;j<15-pl.p[i].name.length();j++)
            {
              s=s+" ";
            }
            s=s+pl.p[i].assist;
            s=s+"      "+pl.p[i].team;
            bw.write (s);
            bw.newLine();
          }
          pl.sort(false,false);
          bw.write ("name           score  team");
          bw.newLine();
          for(int i=0;i<10;i++)
          {
            String s="";
            s=s+pl.p[i].name;
            for(int j=0;j<15-pl.p[i].name.length();j++)
            {
              s=s+" ";
            }
            s=s+pl.p[i].score;
            s=s+"      "+pl.p[i].team;
            bw.write (s);
            bw.newLine();
          }
          bw.write("player       goal assist score");
          bw.newLine();
          for(int i=0;i<8;i++)
          {
            for(int j=0;j<11;j++)
            {
              bw.write(tl.ca[i].member[j].getinf());
              bw.newLine();
            }
            bw.write("score "+tl.ca[i].score+" goal "+tl.ca[i].goal+" number of fans "+tl.ca[i].fan);
            bw.newLine();             
          }
          bw.close();
          tl.transfer();
        }
        
        catch(Exception ee){
            System.out.println(ee);
        }
  }
  public static void writegamerecord(team a,team b)
  {
    try
        {
          BufferedWriter bw = new BufferedWriter(new FileWriter("gamerecord.txt",true));
          bw.write(tostring(tl.year)+ " "+tostring(tl.round)+" "+tostring(tl.game));
          bw.newLine();
          bw.write(a.name+ " "+tostring(a.igoal)+"   "+tostring(b.igoal)+" "+b.name);
          bw.newLine();
          bw.close();
        }
        
        catch(Exception ee){
            System.out.println(ee);
        }
  }
  ////////////////////////////////////////////////////////////////////////write read helper////////////////////////////////////////////////////////
  public static int toint(String s)
  {
    if(s.length()==1)
    {
      return s.charAt(0)-48;
    }
    else
    {
      return (s.charAt(s.length()-1)-48)+10*toint(s.substring(0,s.length()-1));
    }
  }
  public static String tostring(int i)
  {
    return ""+i;
  }
} 

